## Présentation générale du jeu de données
## Démarche méthodologique d’analyse de données 
## Synthèse de l’analyse de données

In [51]:
import pandas as pd
import numpy as np

In [52]:
df = pd.read_csv("data/p2-arbres-fr.csv", delimiter=";", decimal=",")

#### Renaming columns

In [53]:
dict_names = {}
for name in df.columns:
    new_name = name.lower().replace(" ", "_")
    dict_names[name] = new_name
df.rename(columns=dict_names, inplace=True)

### First check of the dataset

In [54]:
df.head()

id type_emplacement domanialite   arrondissement complement_addresse  \
0  99874            Arbre      Jardin   PARIS 7E ARRDT                 NaN   
1  99875            Arbre      Jardin   PARIS 7E ARRDT                 NaN   
2  99876            Arbre      Jardin   PARIS 7E ARRDT                 NaN   
3  99877            Arbre      Jardin   PARIS 7E ARRDT                 NaN   
4  99878            Arbre      Jardin  PARIS 17E ARRDT                 NaN   

   numero                                        lieu id_emplacement  \
0     NaN   MAIRIE DU 7E 116 RUE DE GRENELLE PARIS 7E             19   
1     NaN   MAIRIE DU 7E 116 RUE DE GRENELLE PARIS 7E             20   
2     NaN   MAIRIE DU 7E 116 RUE DE GRENELLE PARIS 7E             21   
3     NaN   MAIRIE DU 7E 116 RUE DE GRENELLE PARIS 7E             22   
4     NaN  PARC CLICHY-BATIGNOLLES-MARTIN LUTHER KING       000G0037   

  libelle_francais      genre         espece variete  circonference_cm  \
0       Marronnier   Aesculus  hippocastanum     NaN                20   
1               If      Taxus        baccata     NaN                65   
2               If      Taxus        baccata     NaN                90   
3           Erable       Acer        negundo     NaN                60   
4     Arbre à miel  Tetradium      daniellii     NaN                38   

   hauteur_m stade_developpement  remarquable geo_point_2d_a geo_point_2d_b  
0          5                 NaN          0.0  48.8576199541   2.3209621099  
1          8                   A          NaN  48.8576560697   2.3210314229  
2         10                   A          NaN  48.8577050069    2.321060566  
3          8                   A          NaN  48.8577220449   2.3210058253  
4          0                 NaN          NaN  48.8904352883   2.3152894588

### Cheking dataset info

In [55]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200137 entries, 0 to 200136
Data columns (total 18 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   id                   200137 non-null  int64  
 1   type_emplacement     200137 non-null  object 
 2   domanialite          200136 non-null  object 
 3   arrondissement       200137 non-null  object 
 4   complement_addresse  30902 non-null   object 
 5   numero               0 non-null       float64
 6   lieu                 200137 non-null  object 
 7   id_emplacement       200137 non-null  object 
 8   libelle_francais     198640 non-null  object 
 9   genre                200121 non-null  object 
 10  espece               198385 non-null  object 
 11  variete              36777 non-null   object 
 12  circonference_cm     200137 non-null  int64  
 13  hauteur_m            200137 non-null  int64  
 14  stade_developpement  132932 non-null  object 
 15  remarquable      

#### Columns with Null Values
##### 2 000 137 raws  

- numero               0 non-null

- complement_addresse  30902 non-null

- libelle_francais     198640 non-null

- genre                200121 non-null

- espece               198385 non-null

- variete              36777 non-null

- stade_developpement  132932 non-null

- remarquable          137039 non-null

- domanialite          200136 non-null

In [57]:
null_col = []
for col in df.columns:
    if True in df[col].isnull().values:
        null_col.append(col)
        

In [62]:
x = df["domanialite"].isnull().values

In [81]:
total = len(df)
values, count = np.unique(x, return_counts=True)
r = dict(zip(values, count))
try:
    r[True]
except: 
    r[False]

200137

##### remove numero column because all values are missing 

In [47]:
df.drop(columns=["numero"], inplace=True)

In [50]:
df[null_col]

domanialite complement_addresse     libelle_francais      genre  \
0           Jardin                 NaN           Marronnier   Aesculus   
1           Jardin                 NaN                   If      Taxus   
2           Jardin                 NaN                   If      Taxus   
3           Jardin                 NaN               Erable       Acer   
4           Jardin                 NaN         Arbre à miel  Tetradium   
...            ...                 ...                  ...        ...   
200132      Jardin                 NaN                  NaN   Castanea   
200133      Jardin                 NaN  Arbre aux mouchoirs    Davidia   
200134      Jardin                 NaN             Peuplier    Populus   
200135      Jardin                 NaN                Chêne    Quercus   
200136      Jardin                 NaN            Raisinier    Hovenia   

               espece variete stade_developpement  remarquable  
0       hippocastanum     NaN                 NaN          0.0  
1             baccata     NaN                   A          NaN  
2             baccata     NaN                   A          NaN  
3             negundo     NaN                   A          NaN  
4           daniellii     NaN                 NaN          NaN  
...               ...     ...                 ...          ...  
200132       fargesii     NaN                 NaN          0.0  
200133    involucrata     NaN                 NaN          0.0  
200134        simonii     NaN                 NaN          0.0  
200135         n. sp.     NaN                 NaN          0.0  
200136         dulcis     NaN                 NaN          0.0  

[200137 rows x 8 columns]